SafetyFirst/ResultModule.fs

```diff
@@ -0,0 +1,467 @@
+  /// to the function given, returning an Ok with the result of that function.  
+  /// If any Result is error, returns the first error without calling the function given. 
+  /// </summary>
+  [<CompilerMessage(message="not for use from F#", messageNumber=32121, IsHidden=true)>]
+  let Map3 (onOk:Func<'a, 'b, 'c, 'd>) result1 result2 result3 = 
+    Bind3 (Func<'a, 'b, 'c, Result<'d, 'e>> (fun r1 r2 r3 -> Ok (onOk.Invoke (r1, r2, r3)))) 
+      result1 result2 result3
+
+  /// <summary>
+  /// If the Result is ok, "unwraps" the ok values and passes them
+  /// to the function given, returning an Ok with the result of that function.  
+  /// If any Result is error, returns the first error without calling the function given. 
+  /// </summary>
+  [<CompilerMessage(message="not for use from F#", messageNumber=32121, IsHidden=true)>]
+  let Map4 (onOk:Func<'a, 'b, 'c, 'd, 'e>) result1 result2 result3 result4 = 
+    Bind4 (Func<'a, 'b, 'c, 'd, Result<'e, 'f>> (fun r1 r2 r3 r4 -> Ok (onOk.Invoke (r1, r2, r3, r4)))) 
+      result1 result2 result3 result4
+
+  /// <summary>
+  /// If the Result is ok, "unwraps" the ok values and passes them
+  /// to the function given, returning an Ok with the result of that function.  
+  /// If any Result is error, returns the first error without calling the function given. 
+  /// </summary>
+  [<CompilerMessage(message="not for use from F#", messageNumber=32121, IsHidden=true)>]
+  let MapAll onOk results = 
+    (concatResultsCs results).Map onOk
+
+  /// <summary>
+  /// Creates a new ok Result with the value given. 
+  /// </summary>
+  [<CompilerMessage(message="not for use from F#", messageNumber=32121, IsHidden=true)>]
+  let Ok<'tVal, 'tErr> v : Result<'tVal, 'tErr> = Ok v
+
+  /// <summary>
+  /// Creates a new error Result with the error object given. 
+  /// </summary>
+  [<CompilerMessage(message="not for use from F#", messageNumber=32121, IsHidden=true)>]
+  let Error<'tVal, 'tErr> v : Result<'tVal, 'tErr> = Error v
+
+  /// <summary>
+  /// Converts an FSharpResult type to a ResultDotNet Result type.  Used for interoping between
+  /// C# and F# codebases. 
+  /// </summary>
+  [<CompilerMessage(message="not for use from F#", messageNumber=32121, IsHidden=true)>]
+  let FromFs v = toCs v
+
+  /// <summary>
+  /// Collects a sequence of Results into a single Result of the sequence of values.
+  /// If all of the Results are Ok, returns an Ok of the sequence of contained values.  
+  /// If any of the Results are Error, returns an Error of the sequence of contained 
+  /// Errors (and throws away any Ok values). 
+  /// <c>Result.Collect [Ok(1), Ok(2), Ok(3)]</c> would return <c>Ok([1, 2, 3])</c>, but 
+  /// <c>Result.Collect [Ok(1), Error("err"), Error("fail")]</c> would return <c>Error(["err", "fail"])</c>
+  /// </summary>
+  [<CompilerMessage(message="not for use from F#", messageNumber=32121, IsHidden=true)>]
+  let Collect results = collect results |> FromFs
```

Comment:
When we switch the `collect` function over to use F# Lists, we should think about this.  The return value should probably be a C# list or an array.

Reply:
Actually, maybe fseq would be best, since that's a type that's still consumable by C#, and wouldn't have a conversion cost.

Reply:
✔️